sycl/include/CL/sycl/detail/generic_type_traits.hpp

```diff
@@ -238,8 +238,9 @@ using is_ulongn = typename is_contained<
 // ugenlong: unsigned long int, ulongn
 template <typename T>
 using is_ugenlong =
-    std::integral_constant<bool, is_contained<T, type_list<cl_ulong>>::value ||
-                                     is_ulongn<T>::value>;
+    std::integral_constant<bool,
+      is_contained<T, type_list<unsigned long, cl_ulong>>::value ||
```

Comment:
According to the comment above, `cl_ulong` is not `ugenlong`.

Also I think we should update the logic of **ALL** traits that are not relying on fixed size types.
We probably don't see these issues now, but we might hit them on some other platforms.
This can be validated by the cross-compilation for the triples, which defines sizes of `char != 1`, `short != 2` and `int != 4`.

Reply:
Good catch. This though does not relate to Melanie's fix, right?
How about fixing that cl_ulong issue and 'update the logic of ALL traits' separately?